### ISSUE TITLE
Support Haskell scripts made with Stack

### ISSUE THREAD

**Issue by pkkm** [flycheck/flycheck#1441]
The Stack build tool has a feature for making self-contained Haskell scripts ([documentation](https://docs.haskellstack.org/en/stable/GUIDE/#script-interpreter)). An example script may be:

```haskell
#!/usr/bin/env stack
{- stack script --resolver lts-11.2 --optimize
     --package attoparsec --package formatting
     -- -Wall -Wextra -}

import qualified Data.Attoparsec.Text as P
import qualified Formatting.Formatters as F
...
```

Currently, Flycheck is not useful for such scripts even with the [flycheck-haskell](https://github.com/flycheck/flycheck-haskell) package. It ignores the script's options and uses global ones, which usually means a different version of GHC and a different set of packages. The result is that the checker shows some spurious errors in the top lines of the file and doesn't check further.

It would be nice if Flycheck could detect that it's checking a Haskell script which uses Stack, then select the Stack-based checker and make it use the options specified below the shebang in the script.

**Comment by fmdkdd** [comment]
Yes, it would be nice indeed.  Because it requires looking at the header, parsing it and passing the options to stack, I think it's best implemented in flycheck-haskell.

WDYT @flycheck/haskell ?

**Comment by sergv** [comment]
Yes, this seems like a job for `flycheck-haskell`. After adding header parameters to `flycheck-ghc-args` they should be picked up by the `haskell-stack-ghc` checker and Just Work™.

**Comment by cpitclaudel** [comment]
Ideally, I don't think we should do the option parsing — doesn't slack itself do it?

**Comment by sergv** [comment]
Well, stack itself certainly does it if we just execute `./Script.hs`. But that would start executing the script, which is likely not what people expect from an Emacs checker. We need to invoke the compiler, `ghc`, with correct options, which we can only infer from that little comment at top of file.

**Comment by cpitclaudel** [comment]
> We need to invoke the compiler, ghc, with correct options, which we can only infer from that little comment at top of file.

Right, I was hoping that slack would offer an option to this.

**Comment by cpitclaudel** [comment]
Moved to https://github.com/flycheck/flycheck-haskell/issues/104
